common/domain/handler.go

```diff
@@ -504,6 +504,7 @@ func (d *handlerImpl) UpdateDomain(
 		now := d.timeSource.Now()
 		// Check the failover cool down time
 		if lastUpdatedTime.Add(d.config.FailoverCoolDown(info.Name)).After(now) {
+			d.logger.Debugf("Domain was last updated at %v, failoverCoolDown: %v, current time: %v.", lastUpdatedTime, d.config.FailoverCoolDown(info.Name), now)
```

Comment:
nit: Maybe even info or warning level. Since we error out, this should not be a hot path; without logs, it is hard to identify the issue without looking into the domain change history.

Reply:
Failover cooldown seems to be working as expected and usually there's no surprise here. I added this to locally make sure my dynamicconfig overrides are working as expected.